ksql-engine/src/main/java/io/confluent/ksql/util/AvroUtil.java

```diff
@@ -64,34 +65,64 @@ public static AbstractStreamCreateStatement checkAndSetAvroSchema(
     }
 
     String kafkaTopicName = StringUtil.cleanQuotes(
-        ddlProperties.get(DdlConfig.KAFKA_TOPIC_NAME_PROPERTY).toString()
+            ddlProperties.get(DdlConfig.KAFKA_TOPIC_NAME_PROPERTY).toString()
     );
+
+    if (!abstractStreamCreateStatement.getElements().isEmpty()) {
+      return abstractStreamCreateStatement;
+    }
+
+    // If the schema is not specified infer it from the Avro schema in Schema Registry.
+    final SchemaMetadata schemaMetadata = getSchemaMetadata(
+            abstractStreamCreateStatement, schemaRegistryClient, kafkaTopicName);
+
     try {
-      // If the schema is not specified infer it from the Avro schema in Schema Registry.
-      if (abstractStreamCreateStatement.getElements().isEmpty()) {
-        SchemaMetadata schemaMetadata = fetchSchemaMetadata(
-            abstractStreamCreateStatement,
-            schemaRegistryClient,
-            kafkaTopicName
-        );
+      final String avroSchemaString = schemaMetadata.getSchema();
+      streamsProperties.put(DdlConfig.AVRO_SCHEMA, avroSchemaString);
+      AbstractStreamCreateStatement abstractStreamCreateStatementCopy = addAvroFields(
+              abstractStreamCreateStatement,
+              AvroSchemaTranslator.toKsqlSchema(avroSchemaString),
+              schemaMetadata.getId()
+      );
+      return abstractStreamCreateStatementCopy;
 
-        String avroSchemaString = schemaMetadata.getSchema();
-        streamsProperties.put(DdlConfig.AVRO_SCHEMA, avroSchemaString);
-        AbstractStreamCreateStatement abstractStreamCreateStatementCopy = addAvroFields(
-            abstractStreamCreateStatement,
-            AvroSchemaTranslator.toKsqlSchema(avroSchemaString),
-            schemaMetadata.getId()
-        );
-        return abstractStreamCreateStatementCopy;
-      } else {
-        return abstractStreamCreateStatement;
-      }
-    } catch (Exception e) {
-      String errorMessage = String.format(
-          " Unable to verify the AVRO schema is compatible with KSQL. %s ",
-          e.getMessage()
+    } catch (final Exception e) {
+      throw new KsqlException("KSQL does not currently support the Avro schema for topic "+kafkaTopicName+
```

Comment:
There may be different kinds of errors thrown here (e.g. some sort of network error). Also, with this PR (https://github.com/confluentinc/ksql/pull/1529) we wouldn't fail with this error. Instead, types that aren't supported will be omitted from the ksql schema. So maybe keep this one as is?

Reply:
I've reworded it